Framework/Algorithms/src/GeneratePeaks.cpp

```diff
@@ -767,7 +767,7 @@ GeneratePeaks::createDataWorkspace(std::vector<double> binparameters) {
   MatrixWorkspace_sptr ws = API::WorkspaceFactory::Instance().create(
       "Workspace2D", m_spectraSet.size(), numxvalue, numxvalue - 1);
   for (size_t ip = 0; ip < m_spectraSet.size(); ip++)
-    std::copy(xarray.begin(), xarray.end(), ws->dataX(ip).begin());
+    std::copy(xarray.begin(), xarray.end(), ws->mutableX(ip).begin());
```

Comment:
Since we have the same `X` for all spectra it would be better to set up sharing. From the way `ws` is created you can see that it is storing `BinEdges`, so here we should simply write

```cpp
ws->setBinEdges(ip, edges);
```
Where `edges` is should be created before the loop.